application/power-management/default.nix

```diff
@@ -0,0 +1,104 @@
+{ pkgs, config, lib, ... }:
+
+let
+
+  user = "power-button-shutdown";
+  group = "power-button-shutdown";
+
+  power-button-shutdown = pkgs.stdenv.mkDerivation {
+    name = "power-button-shutdown";
+    propagatedBuildInputs = [
+      (pkgs.python3.withPackages (pythonPackages: with pythonPackages; [
+        evdev
+      ]))
+    ];
+    dontUnpack = true;
+    installPhase = "install -Dm755 ${./power-button-shutdown.py} $out/bin/power-button-shutdown";
+  };
+
+in {
+
+  # Ignore system control keys that do not make sense for kiosk applications.
+  # Ignore power key as well, but set up a systemd service shutting down the
+  # computer on Power Button key press.
+  services.logind.extraConfig = ''
+    HandlePowerKey=ignore
+    HandleRebootKey=ignore
+    HandleSuspendKey=ignore
+    HandleHibernateKey=ignore
+    HandlePowerKeyLongPress=poweroff
+    HandleRebootKeyLongPress=poweroff
+    HandleSuspendKeyLongPress=poweroff
+    HandleHibernateKeyLongPress=poweroff
+  '';
+
+  users = {
+    users.${user} = {
+      description = "User executing power-button-shutdown.service";
+      group = group;
+      createHome = false;
+      isSystemUser = true;
+    };
+    groups.${group} = {};
+  };
+
+  hardware.uinput.enable = lib.mkDefault true;
```

Comment:
Is this needed?

I made a build with this modification, and the built-in power button still works:

```diff
diff --git a/application/power-management/default.nix b/application/power-management/default.nix
index f6827e7..c74c5bd 100644
--- a/application/power-management/default.nix
+++ b/application/power-management/default.nix
@@ -42,8 +42,6 @@ in {
     groups.${group} = {};
   };
 
-  hardware.uinput.enable = lib.mkDefault true;
-
   # Allow user of power-button-shutdown.service to shutdown the service
   security.polkit = {
     enable = true;
@@ -70,7 +68,7 @@ in {
 
       User = user;
       Group = group;
-      SupplementaryGroups = with config.users.groups; [ input.name uinput.name ];
+      SupplementaryGroups = with config.users.groups; [ input.name ];
 
       # Hardening, see https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/hardware/kanata.nix#L117 for inspiration.
       # Not using DeviceAllow and DevicePolicy, as this prevent listing devices otherwise.

```